src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -60,7 +60,7 @@ public bool CanNavigateToSpan(Workspace workspace, DocumentId documentId, TextSp
             }
 
             var document = workspace.CurrentSolution.GetDocument(documentId);
-            var text = document.GetTextSynchronously(CancellationToken.None);
+            var text = document.GetTextSynchronously(cancellationToken);
```

Comment:
oh dear lord...

Reply:
So this is "fine" in that this code is only running if it's a secondary buffer; in that case the file is open and GetTextSynchronously is doing absolutely nothing exciting.

Reply:
Now that said, the bounds check two lines past this is terrifying _me_, so we can at least agree this code is terrifying.